### PR TITLE
Redux

### PR DESCRIPTION
**Pull request by justindwyer6**
This PR adds Redux and implements it for the rounds object (now called "realms") and the rules toggle.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,13 +1,19 @@
...
+const store = createStore();
 
 ReactDOM.render(
   <React.StrictMode>
     <ErrorBoundary>
-      <App />
+      <Provider store={store}>
+        <App />
+      </Provider>
     </ErrorBoundary>
```
Comment (abrad45) [inline src/index.js:17]: Not a problem or request, but as we talk about boilerplate, if you wanted to, all your additions here could totally be put in some other file:

```js
// provider.js
import { Provider } from "react-redux";
import createStore from "./reducers";

const store = createStore();

export const ReduxProvider = ({ children }) => {
  return (
    <Provider store={store}>
      {children}
    </Provider>
  );
}
```

If you did, in this file you'd need simply to import and then wrap `<App />` in `<ReduxProvider>`.

Does it save anything? No. But if you added more complex logic in redux, like say persisting your redux store in local storage, the file might end up being more redux than app. This encapsulation can help, but might also be overengineering now :)

File: src/components/App/App.js [hunk 2]
```diff
@@ -7,50 +8,42 @@ import IconButton from "../IconButton/IconButton";
...
+    realms,
+    setRealms,
+  } = useSetRealms();
+  const toggleRules = useToggleRules();
+  const areRulesOpen = useAreRulesOpen();
 
   return (
     <div className="appContainer">
-      <Rules rulesOpen={rulesOpen} setRulesOpen={setRulesOpen} />
+      <Rules rulesOpen={areRulesOpen} setRulesOpen={toggleRules} />
```
Comment (abrad45) [inline src/components/App/App.js:40]: Does `Rules` do anything with `setRulesOpen`? I seem to think we decided it does not. If not, let's not pass things in that don't get used.

**Review by abrad45 — APPROVED** [review]
This looks really good. Most of my comments here were FYIs and not requests. Well done :)

File: src/index.js (hunk 1 above)
Comment (justindwyer6) [reply src/index.js:17]: Awesome, thanks! I'll keep it in mind.

File: src/realms.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  "Scythe",
+  "Between Two Cities",
+  "Charterstone",
+  "Between Two Castles",
+  "Viticulture",
+  "Euphoria",
+  "My Little Scythe",
+  "Tapestry",
+  "Wingspan",
+];
```
[the remark this answers is not in the document]
Comment (justindwyer6) [reply src/realms.js:11]: Adding this as an item in the project kanban.

**Pull request merged**